src/codegen/mod.rs

```diff
@@ -637,12 +635,10 @@ impl CodeGenerator for Type {
                     return;
                 }
 
-                let mut outer_params = item.used_template_params(ctx)
-                    .and_then(|ps| if ps.is_empty() {
-                        None
-                    } else {
-                        Some(ps)
-                    });
+                let mut outer_params = match item.used_template_params(ctx).len() {
```

Comment:
This should probably be just:

```rust
let used_params = item.used_template_params(ctx);
let outer_params = if used_params.is_empty() { None } else { Some(used_params) };
```

Or something of the sort.

Reply:
Yes!, that's cleaner, thanks! 😄